internal/bminventory/inventory.go

```diff
@@ -338,7 +338,7 @@ func (b *bareMetalInventory) setDefaultRegisterClusterParams(_ context.Context,
 func (b *bareMetalInventory) validateRegisterClusterInternalParams(params *installer.V2RegisterClusterParams, log logrus.FieldLogger) error {
 	var err error
 
-	if err = validateProxySettings(params.NewClusterParams.HTTPProxy,
+	if err = validateClusterProxySettings(params.NewClusterParams.HTTPProxy,
```

Comment:
Unsure why we need an additional function for validateClusterProxySettings.
The openshift version is required in NewClusterParams you can assume it's not nil.

Reply:
I added this specifically for cluster because it was checked for in master:
https://github.com/openshift/assisted-service/blob/6751dafe5fe0dbcf4a475c62b4fadf1a69b24165/internal/bminventory/inventory.go#L3800-L3803

However, you make a valid point, it will be enforced anyhow by:


https://github.com/openshift/assisted-service/blob/6751dafe5fe0dbcf4a475c62b4fadf1a69b24165/swagger.yaml#L4172-L4177

Changing this.

Reply:
done